sdk/keyvault/azure-security-keyvault-shared/inc/azure/keyvault/shared/keyvault_challenge_based_auth.hpp

```diff
@@ -35,7 +47,8 @@ namespace Azure { namespace Security { namespace KeyVault { namespace _internal
 
     std::unique_ptr<HttpPolicy> Clone() const override
     {
-      return std::make_unique<KeyVaultChallengeBasedAuthenticationPolicy>(*this);
+      return std::unique_ptr<KeyVaultChallengeBasedAuthenticationPolicy>(
```

Comment:
Please add a comment explaining why you can't simply call `std::make_unique` here. Otherwise someone like me will come along and decide that the code can be made simpler by changing it to a call to `std::make_unique`.